pkg/sql/sem/tree/expr.go

```diff
@@ -1374,11 +1374,10 @@ func (node *FuncExpr) Format(ctx *FmtCtx) {
 		typ = funcTypeName[node.Type] + " "
 	}
 
-	// We need to remove name anonymization/redaction for the function name in
-	// particular. Do this by overriding the flags.
-	// TODO(thomas): when function names are correctly typed as FunctionDefinition
-	// remove FmtMarkRedactionNode from being overridden.
-	ctx.WithFlags(ctx.flags&^FmtAnonymize&^FmtMarkRedactionNode|FmtBareIdentifiers, func() {
+	// Ideally we would not anonymize/redact built-in function names. Some
+	// built-in functions are correctly typed as FunctionDefinition and so avoid
+	// redaction.
+	ctx.WithFlags(ctx.flags|FmtBareIdentifiers, func() {
```

Comment:
not blocking this PR, but i wonder if it's true that `node.fnProps` is only non-nil for builtin functions. that could be a way to determine if the name needs to be redacted

Reply:
`fnProps` is written to during type-checking when the overload is resolved. At that point we can determine if the function is built-in or not, using `Overload.Type == BuiltinRoutine`. But if the AST is printed before being type-checked, we cannot.